### PR TITLE
Refactor: declare pdf whitespaces in global variable to avoid bug

### PR DESCRIPTION
**Pull request by LucianoHanna**
Hi,

I made some refactor in RawDataParser, I declared a global variable containing pdf whitespaces so we don't need to hardcode were use whitespace, avoiding bug if forget to consider all pdf whitespaces

### CODE REVIEW

File: src/Smalot/PdfParser/RawData/RawDataParser.php [hunk 1]
```diff
@@ -570,16 +585,8 @@ protected function getRawObject($pdfData, $offset = 0)
...
-         * skip initial white space chars:
-         *      \x00 null (NUL)
-         *      \x09 horizontal tab (HT)
-         *      \x0A line feed (LF)
-         *      \x0C form feed (FF)
-         *      \x0D carriage return (CR)
-         *      \x20 space (SP)
-         */
-        $offset += strspn($pdfData, "\x00\x09\x0a\x0c\x0d\x20", $offset);
+        //skip initial white space chars
```
Comment (k00ni) [inline src/Smalot/PdfParser/RawData/RawDataParser.php:588]: Please add a whitespace before skip.

File: src/Smalot/PdfParser/RawData/RawDataParser.php [hunk 2]
```diff
@@ -57,6 +57,10 @@ class RawDataParser
     protected $filterHelper;
     protected $objects;
 
+    // (NUL, HT, LF, FF, CR, SP)
+    protected $pdfWhitespaces = "\0\t\n\f\r ";
```
Comment (k00ni) [inline src/Smalot/PdfParser/RawData/RawDataParser.php:61]: Please make it a comment as of:

```php
/**
 * Represents: (NUL, HT, LF, FF, CR, SP)
 */
```

File: src/Smalot/PdfParser/RawData/RawDataParser.php [hunk 3]
```diff
@@ -486,18 +503,16 @@ protected function getIndirectObject($pdfData, $xref, $objRef, $offset = 0, $dec
         if (2 !== \count($objRefArr)) {
             throw new Exception('Invalid object reference for $obj.');
         }
-        $objHeader = $objRefArr[0].' '.$objRefArr[1].' obj';
 
+        $objHeaderLen = $this->getObjectHeaderLen($objRefArr);
         /*
```
Comment (k00ni) [inline src/Smalot/PdfParser/RawData/RawDataParser.php:508]: Add new line in between.

**Review by k00ni — CHANGES_REQUESTED** [review]
@LucianoHanna thank you for the PR.

I welcome your changes in general but have to ask you to move `$pdfWhitespaces` and `$pdfWhitespacesRegex` to [Config.php](https://github.com/smalot/pdfparser/blob/master/src/Smalot/PdfParser/Config.php). 

Its purpose is to hold default values which can be overridden by the user. In some cases it is required to change a default value because a given set of PDFs require it. [Here](https://github.com/smalot/pdfparser/blob/ce6aa779dd67fadf3887dace6f31ea70882dba73/tests/Integration/ParserTest.php#L271-L282) is a basic example how to setup and use `Config` instance.

**Comment by k00ni** [comment]
Another remark: Please create a separate branch when creating a pull request. It limits our options if you push using your `master`.

File: src/Smalot/PdfParser/Config.php [hunk 4]
```diff
@@ -56,22 +56,22 @@ public function setFontSpaceLimit($value)
...
         $this->pdfWhitespaces = $pdfWhitespaces;
     }
 
-    public function getPdfWhitespacesRegex(): string
+    public function getPdfWhitespacesRegex()
     {
         return $this->pdfWhitespacesRegex;
     }
 
-    public function setPdfWhitespacesRegex(string $pdfWhitespacesRegex): void
```
Comment (k00ni) [inline src/Smalot/PdfParser/Config.php]: We merged in #383 lately, which means we only support 7.1+. Support for earlier PHP versions is not required anymore. I like these explicit return values, can you re-add them please?

**Review by k00ni — CHANGES_REQUESTED** [review]

File: src/Smalot/PdfParser/RawData/RawDataParser.php [hunk 5]
```diff
@@ -463,6 +470,19 @@ protected function decodeXrefStream($pdfData, $startxref, $xref = [])
...
+        // consider all whitespace character (PDF specifications)
+        return '/'.$objRefArr[0].$this->config->getPdfWhitespacesRegex().$objRefArr[1].$this->config->getPdfWhitespacesRegex().'obj'.'/';
+    }
+
+    protected function getObjectHeaderLen($objRefArr)
+    {
+        // "4 0 obj"
+        // 2 whitespaces + strlen("obj") = 5
+        return 5 + \strlen($objRefArr[0]) + \strlen($objRefArr[1]);
+    }
```
Comment (k00ni) [inline src/Smalot/PdfParser/RawData/RawDataParser.php:484]: Can you add `: int` for each function to make return value type more clear?

File: src/Smalot/PdfParser/Config.php [hunk 6]
```diff
@@ -40,6 +40,12 @@ class Config
 {
     private $fontSpaceLimit = -50;
 
+    /**
+     * Represents: (NUL, HT, LF, FF, CR, SP)
+     */
+    private $pdfWhitespaces = "\0\t\n\f\r ";
+    private $pdfWhitespacesRegex = '[\0\t\n\f\r ]';
```
Comment (k00ni) [inline src/Smalot/PdfParser/Config.php:47]: Can you add `@var string` here for both variables too please?

**Comment by LucianoHanna** [comment]
@k00ni I made these changes but there's one test failing, maybe should open another PR to remove "CI / Tests alternative autoloader (PHP 5.6)"

**Comment by k00ni** [comment]
#409 already targets the outdated alternative autoloader tests. Can you merge it and try again? In the end I will squash your PR, so it shouldn't matter

**Review by k00ni — APPROVED** [review]
👍🏿

**Comment by k00ni** [comment]
We had to fix some problems in our testing environment (Github workflow + Scrutinizer). All fixes are now in master. Can you please merge `master` to see if there are any problems left?

**Comment by LucianoHanna** [comment]
I merged master and everything looks ok but there's nothing left to commit...

**Comment by k00ni** [comment]
There should be a merge commit to push. Anyway.

**Pull request merged**